### ISSUE TITLE
Core cannot login to internally provided postgres db after upgrade from 2.8.0 to 2.10.0 (using helm charts)

### ISSUE THREAD

**Issue by peter-englmaier** [goharbor/harbor-helm#1683]
After the upgrade to 2.10.0, core and jobservice did not start anymore. In the core output, I see that it cannot log into database:

```
2024-01-16T11:01:34Z [INFO] [/common/dao/base.go:67]: Registering database: type-PostgreSQL host-harbor2-database port-5432 database-registry sslmode-"disable"
[ORM]2024/01/16 11:01:34 register db Ping `default`, failed to connect to `host=harbor2-database user=postgres database=registry`: failed SASL auth (FATAL: password authentication failed for user "postgres" (SQLSTATE 28P01))
2024-01-16T11:01:34Z [FATAL] [/core/main.go:184]: failed to initialize database: register db Ping `default`, failed to connect to `host=harbor2-database user=postgres database=registry`: failed SASL auth (FATAL: password authentication failed for user "postgres" (SQLSTATE 28P01))
Log
```
However, with the password from the environment variable `POSTGRES_PASSWORD`, I can login to the `postgres` user from pods running in the cluster. 

**Expected behavior and actual behavior:**
Upgrade should work with the provided build-in postgres DB.

**Steps to reproduce the problem:**
This might only happen with newer k8s versions (1.29.0) or GKE.

**Versions:**
- harbor version: 2.10.0
- kubernetes: Google GKE version1.29.0-gke.1381000
- docker engine version: containerd 1.7.10
- helm-chart: https://helm.goharbor.io/harbor-1.14.0
- helm version: v3.13.3

**Additional context:**

A simple hotfix for the issue is, to edit the secrets generated by the helm chart (after install) with `kubectl edit secret`. The two secrets I found problematic are `${HELMRELEASE}-database` and `${HELMRELEASE}-core`. Just edit and replace the values for `POSTGRES_PASSWORD` and `POSTGRESQL_PASSWORD` with base64 encoded strings. Restart the deployment `kubectl rollout restart deploy` and it should now start.

Perhaps this is an GKE  or k8s 1.29 issue. When looking inside the POSTGRES Pod, the environment says something binary:
```
POSTGRES_PASSWORD=r�9
```
Since the DB already existed, the value is ignored here. In the core POD, however, it means the wrong password is used.

**Comment by kengibous** [comment]
it does not seem to be base 64 encoding the database password

For example leaving it as the default "changeit"

Should result in `Y2hhbmdlaXQ=` in the `${HELMRELEASE}-database` and `${HELMRELEASE}-core` but shows as `changeit` - i confirmed that my 2.9.1 version has the proper base64 encoded value but upgrading to 2.10 changes it not be base64 encoded

**Comment by kengibous** [comment]
@peter-englmaier - if the secrets already exist `<release name>-database` and `<release name>-core` doing a `helm upgrade` will result in the values showing in the clear and not properly base64 encoded.  I deleted both of those secrets and them did a `helm upgrade` and the behavior was correct.  This issue mentions it - https://github.com/goharbor/harbor-helm/issues/1674 and a fix that was merged https://github.com/goharbor/harbor-helm/pull/1679/files.  A future release of the chart is ultimately needed.
